Trying to build BDN on a windows arm64 host

To address the issue I mentioned in #962 where we're not getting the right native dependencies for arm64 runs to enable ETW.

I have only an arm64 dotnet installed globally. Another one gets installed under .dotnet.

```
C:\repos\BenchmarkDotNet>.\.dotnet\dotnet.exe --info
.NET SDK (reflecting any global.json):
 Version:   6.0.201
 Commit:    ef40e6aa06

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-arm64
 Base Path:   C:\repos\BenchmarkDotNet\.dotnet\sdk\6.0.201\
```

First problem I hit was that the build\Build.proj is asking for netcoreapp3.1, which isn't supported for arm64. So this seems to build for 
x64 instead, and then fails to load hostxfer.dll from the .dotnet as that is an arm64 dll. 
```
--------------------------------------------------------------------------------------
Failed to load the dll from [C:\repos\BenchmarkDotNet\.dotnet\host\fxr\6.0.3\hostfxr.dll], HRESULT: 0x800700C1
The library hostfxr.dll was found, but loading it from C:\repos\BenchmarkDotNet\.dotnet\host\fxr\6.0.3\hostfxr.dll failed
  - Installing .NET Core prerequisites might help resolve this problem.
 
(from event log)

Description: A .NET Core application failed.
Application: Build.exe
Path: C:\repos\BenchmarkDotNet\build\bin\Debug\netcoreapp3.1\Build.exe
Message: Failed to load the dll from [C:\repos\BenchmarkDotNet\.dotnet\host\fxr\6.0.3\hostfxr.dll], HRESULT: 0x800700C1
The library hostfxr.dll was found, but loading it from C:\repos\BenchmarkDotNet\.dotnet\host\fxr\6.0.3\hostfxr.dll failed
 ```

Updating the build target to net6.0 gets me past this.

But then a bunch of tests also fail with similar cross-arch errors:
```
  BenchmarkDotNet.IntegrationTests.EnabledOptimizations -> C:\repos\BenchmarkDotNet\tests\BenchmarkDotNet.IntegrationTests.EnabledOptimizations\bin\Release\net6.0\BenchmarkDotNet.IntegrationTests.EnabledOptimizations.dll
CSC : error CS0041: Unexpected error writing debug information -- 'An attempt was made to load a program with an incorrect format. (0x8007000B)' [C:\repos\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj]
CSC : error CS0041: Unexpected error writing debug information -- 'An attempt was made to load a program with an incorrect format. (0x8007000B)' [C:\repos\BenchmarkDotNet\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj]
```
so presumably fixes are also needed somewhere else.

--- a/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj
+++ b/samples/BenchmarkDotNet.Samples/BenchmarkDotNet.Samples.csproj
@@ -9,7 +9,6 @@
     <PackageId>BenchmarkDotNet.Samples</PackageId>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
     <DebugSymbols>true</DebugSymbols>
     <NoWarn>$(NoWarn);CA1018;CA5351</NoWarn>
   </PropertyGroup>

--- a/src/BenchmarkDotNet/Templates/CsProj.txt
+++ b/src/BenchmarkDotNet/Templates/CsProj.txt
@@ -11,7 +11,7 @@
     <OutputType>Exe</OutputType>
     <OutputPath>bin\$CONFIGURATIONNAME$</OutputPath>
     <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
-    <!-- disabled due to https://github.com/dotnet/roslyn/issues/59240 -->
+    <!-- disabled due to https://github.com/dotnet/roslyn/issues/59421 -->
     <DebugSymbols>false</DebugSymbols>
     <UseSharedCompilation>false</UseSharedCompilation>
     <CodeAnalysisRuleSet></CodeAnalysisRuleSet>

--- a/tests/BenchmarkDotNet.IntegrationTests.ManualRunning/BenchmarkDotNet.IntegrationTests.ManualRunning.csproj
+++ b/tests/BenchmarkDotNet.IntegrationTests.ManualRunning/BenchmarkDotNet.IntegrationTests.ManualRunning.csproj
@@ -9,7 +9,6 @@
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <ItemGroup>

--- a/tests/BenchmarkDotNet.IntegrationTests/BenchmarkDotNet.IntegrationTests.csproj
+++ b/tests/BenchmarkDotNet.IntegrationTests/BenchmarkDotNet.IntegrationTests.csproj
@@ -9,7 +9,6 @@
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
     <DebugSymbols>true</DebugSymbols>
     <NoWarn>$(NoWarn);CA2007</NoWarn>
     <ServerGarbageCollection>true</ServerGarbageCollection>

